### PR TITLE
Remove `SyntaxKind` conformance to `CaseIterable`

### PR DESCRIPTION
**Pull request by ahoppen**
There’s no good reason to iterate over all cases in `SyntaxKind` and if `SyntaxKind` conforms to `CaseIterable`, we can’t mark members of it as deprecated (https://github.com/apple/swift-syntax/pull/2237/files#r1359917461)

While doing this I noticed that `IntegerLiteralExprSyntax.Radix` also conformed to `CaseIterable`. Similarly, there’s no good reason to iterate over all radix kinds, so I removed that conformance as well.

### CODE REVIEW

**Review by kimdv — APPROVED** [review]

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by rintaro — APPROVED** [review]

**Comment by rintaro** [comment]
@swift-ci Please test

File: Release Notes/511.md [hunk 1]
```diff
@@ -16,6 +16,12 @@
 - Effect specifiers:
   - Description: The `unexpectedAfterThrowsSpecifier` node of the various effect specifiers has been removed.
   - Pull request: https://github.com/apple/swift-syntax/pull/2219
+- `SyntaxKind` removed conformance to `CaseIterable`
+  - Description: `SyntaxKind` no longer conforms to `CaseIterable` since there is no good use case to iterate over all syntax kinds. 
+  - Pull request:
```
Comment (Matejkob) [inline Release Notes/511.md:21]: I'm not sure if you've forgotten or are planning to do this next, but the Release Notes are missing the PR link. Here's the suggestion:
```suggestion
  - Pull request: https://github.com/apple/swift-syntax/pull/2292
```

File: Release Notes/511.md [hunk 2]
```diff
@@ -16,6 +16,12 @@
 - Effect specifiers:
   - Description: The `unexpectedAfterThrowsSpecifier` node of the various effect specifiers has been removed.
   - Pull request: https://github.com/apple/swift-syntax/pull/2219
+- `SyntaxKind` removed conformance to `CaseIterable`
+  - Description: `SyntaxKind` no longer conforms to `CaseIterable` since there is no good use case to iterate over all syntax kinds. 
+  - Pull request: 
+- `IntegerLiteralExprSyntax.Radix` removed conformance to `CaseIterable`
+  - Description: `IntegerLiteralExprSyntax.Radix` no longer conforms to `CaseIterable` since there is no good use case to iterate over all radix kinds. 
+  - Pull request:
```
Comment (Matejkob) [inline Release Notes/511.md:24]: And simular here:
```suggestion
  - Pull request: https://github.com/apple/swift-syntax/pull/2292
```

Comment (ahoppen) [reply Release Notes/511.md:24]: I did indeed forget to update it. Thanks for reminding me.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
@swift-ci please test windows

**Pull request merged**